### PR TITLE
Membership tab polish

### PR DESCRIPTION
**Pull request by feedmypixel**
- Tidy up of Styles, Markup, JavaScript, SVGs
  - split membership css into separate file
  - tidy up css and selector names
  - tidy up of JavaScript methods and pseudo constants and removal of componentJS
  - change to how the membershipTab is initialised
  - clean up of markup and move markup into fragments
  - remove card SVGs that we don't accept
- Make the membership tab display at all times
- Provide up sell when Identity user is not a member (first cut - more designs to come)
- Load member details (as before) when Identity user is a Guardian Member
- Provide loading gif when collecting data from membership /user/me endpoint
- Improve qa- hooks for testing
- Added TODO to swap out the old payment form work with the recently refactored [form.js](https://github.com/guardian/membership-frontend/pull/147) work on membership
## Membership Tab (always shown)

screen shot 2015-02-10 at 14 41 49
## Membership Tab (no Tier)

notier
## Membership Tab (with  Tier)

tier

### CODE REVIEW

**Comment by feedmypixel** [comment]
@davidrapson @jamesoram please have look over

**Comment by jamesoram** [comment]
:+1: but unfortunately is still broken in IE9

**Comment by feedmypixel** [comment]
> but unfortunately is still broken in IE9

ok but not a regression - I have added this pr to Jira

File: identity/app/views/fragments/profile/membershipForm/upsell.scala.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div class="membership-up-sell is-hidden js-mem-up-sell">
+    <div class="membership-up-sell__heading">
+        <h2 class="membership-header">Join Guardian Membership</h2>
+    </div>
+    <p class="membership-up-sell__content">We're bringing the Guardian to life through live events and meet-ups Join and be part of the conversations that matter</p>
```
Comment (mattandrews) [inline identity/app/views/fragments/profile/membershipForm/upsell.scala.html]: Looks to me like "Join" should be on a new line?

File: static/src/stylesheets/module/identity/_membership-tab.scss [hunk 2]
```diff
@@ -0,0 +1,272 @@
+/* Edit profile Membership tab
+   ========================================================================== */
+.membership-tab {
+    padding-bottom: $gs-baseline*2;
+
+    .show-updating {
+        display: inline-block;
+    }
+
+    .membership-loader {
```
Comment (davidrapson) [inline static/src/stylesheets/module/identity/_membership-tab.scss]: Feels like a slightly awkward level of nesting here, and only slightly using BEM classnames, but it seems similar to the rest of the CSS so probably not worth changing too much in this PR.

File: static/src/javascripts/projects/membership/membership-tab.js [hunk 3]
```diff
@@ -3,83 +3,64 @@ define([
...
     }
 
-    function formatDate(timestamp) { // eg: 4th June 2014
+    function formatDate(timestamp) {
         var date = new Date(timestamp),
             months = [
-                'January', 'Feburary', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October',
-                'November', 'December'
+                'January',
+                'Feburary',
```
Comment (mattandrews) [inline static/src/javascripts/projects/membership/membership-tab.js]: Typo! Should be "February".

**Comment by mattandrews** [comment]
:+1: besides the copy issues.

**Comment by davidrapson** [comment]
:+1: Looks good, nice bit of cleanup.

File: static/src/stylesheets/module/identity/_membership-tab.scss (hunk 2 above)
Comment (feedmypixel) [inline static/src/stylesheets/module/identity/_membership-tab.scss]: > Feels like a slightly awkward level of nesting here, and only slightly using BEM classnames, but it seems similar to the rest of the CSS so probably not worth changing too much in this PR.

this is for name spacing the css so there isn't a clash with .com work - Agreed this is not ideal but I felt this was preferable to the previous way the css was being name spaced by adding "membership-tab" to selectors

File: identity/app/views/fragments/profile/membershipForm/upsell.scala.html (hunk 1 above)
Comment (feedmypixel) [inline identity/app/views/fragments/profile/membershipForm/upsell.scala.html]: > Looks to me like "Join" should be on a new line?

according to @davidmcdowell it isn't - will confirm with him when he's back
screen shot 2015-02-11 at 15 16 38

Comment (mattandrews) [inline identity/app/views/fragments/profile/membershipForm/upsell.scala.html]: That must be a mistake then (copy/paste fail) – that's definitely two sentences. A full stop after "meet-ups" would work if it all has to be one line.

Comment (feedmypixel) [inline identity/app/views/fragments/profile/membershipForm/upsell.scala.html]: good point - adding in.

**Pull request merged**